ampd/src/sui/verifier.rs

```diff
@@ -133,20 +73,21 @@ impl PartialEq<&VerifierSetConfirmation> for &SuiEvent {
             .chain(expected.created_at.to_be_bytes())
             .collect::<Vec<_>>();
 
-        match serde_json::from_value::<SignersRotated>(self.parsed_json.clone()) {
+        match bcs::from_bytes::<SignersRotated>(&self.bcs) {
             Ok(SignersRotated {
                 signers:
                     WeightedSigners {
                         mut signers,
                         threshold,
                         nonce,
                     },
+                ..
             }) => {
                 signers.sort();
```

Comment:
they're expected to be in sorted order already in the event. It would be concerning if they're not. Perhaps avoid this?

Reply:
Don't think that's relevant to the context of PartialEq. That's external knowledge that this function should not have known. Testing for that ordering is of the job of the contracts.

Reply:
Hmm, while it wouldn't matter for a generic PartialEq implementation, in this case we're trying to validate the event. It would be concerning if the event order isn't correct since the verifier set hash computation changes between internal/external gateway.  Maybe the hash should be checked explicitly. This is more of a sanity check I suppose.